newton/tests/test_viewer_get_frame.py

```
@@ -310,5 +310,50 @@ def test_cpu_viewer_uses_host_pbo_readback(self):
+            granted = viewer.renderer.window.config.samples
+            # A driver offering no multi-sampled config at all reports 0; any
+            # other value has to agree with what the scene is rendered at.
+            self.assertIn(granted, (0, viewer.renderer.msaa_samples))
+        finally:
+            viewer.close()
+
+    def test_scene_keeps_msaa_without_a_multisampled_window_config(self):
+        """Verify the scene stays anti-aliased when the window config has no MSAA."""
+        try:
+            import pyglet.window
+        except Exception as exc:
```

Ruff (0.16.2)

[warning] Do not catch blind exception: `Exception`

(BLE001)